src/Constants.ts

```diff
@@ -13,6 +13,11 @@ export const PBKDF2_ITERATIONS = () => 250000; //Set this as a function so that
  */
 export const DOCUMENT_ENCRYPTION_DETAILS_VERSION_NUMBER = 2;
 
+/**
+ * Regex of allowed characters that users can provide as part of user, group, and document IDs.
+ */
+export const ALLOWED_ID_CHAR_REGEX = /^[a-zA-Z0-9_.$#|@/:;=+'-]+$/;
```

Comment:
If you wanted to be more specific, it's actually `^[a-zA-Z0-9_.$#|@/:;=+'-]{1,100}$`

Reply:
Done, also added a length test.